### PR TITLE
Added a feature, `show_disks` to the `MDFileManager` class, that allows you to display the disks and folders contained in them

### PR DESCRIPTION
**Pull request by Neizvestnyj**
Added a feature that allows you to show the available disks first, then the files contained in them. Works correctly on: **Windows**, **Linux**, **OSX**, **Android**, not tested on **iOS**. A video demonstration can be viewed [here](https://discord.com/channels/566880874789076992/606764006425755648/853298683760214066).
If there are any suggestions and tips, I will take note and change the commit.

```python
from kivy.core.window import Window
from kivy.lang import Builder

from kivymd.app import MDApp
from kivymd.uix.filemanager import MDFileManager
from kivymd.toast import toast

KV = '''
BoxLayout:
    FloatLayout:

        MDRoundFlatIconButton:
            text: "Open manager"
            icon: "folder"
            pos_hint: {'center_x': .5, 'center_y': .6}
            on_release: app.file_manager_open()
'''


class Example(MDApp):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        Window.bind(on_keyboard=self.events)
        self.manager_open = False
        self.file_manager = MDFileManager(
            exit_manager=self.exit_manager,
            select_path=self.select_path,
        )

    def build(self):
        return Builder.load_string(KV)

    def file_manager_open(self):
        self.file_manager.show_disks()  # output manager to the screen
        self.manager_open = True

    def select_path(self, path):
        '''It will be called when you click on the file name
        or the catalog selection button.

        :type path: str;
        :param path: path to the selected directory or file;
        '''

        self.exit_manager()
        toast(path)

    def exit_manager(self, *args):
        '''Called when the user reaches the root of the directory tree.'''

        self.manager_open = False
        self.file_manager.close()

    def events(self, instance, keyboard, keycode, text, modifiers):
        '''Called when buttons are pressed on the mobile device.'''

        if keyboard in (1001, 27):
            if self.manager_open:
                self.file_manager.back()
        return True


Example().run()
```

### CODE REVIEW

File: kivymd/uix/filemanager.py [hunk 1]
```diff
@@ -120,7 +120,8 @@ def events(self, instance, keyboard, keycode, text, modifiers):
 __all__ = ("MDFileManager",)
 
 import locale
-import os
+import os, re
```
Comment (HeaTTheatR) [inline kivymd/uix/filemanager.py]: PEP8:

```python
import os
import re
```

File: kivymd/uix/filemanager.py [hunk 2]
```diff
@@ -469,6 +472,50 @@ def sort_by_name(files):
 
         return sorted_files
 
+    def show_disks(self):
+        if platform == 'win':
```
Comment (HeaTTheatR) [inline kivymd/uix/filemanager.py]: Use `"` instead `'`

File: kivymd/uix/filemanager.py [hunk 3]
```diff
@@ -469,6 +472,50 @@ def sort_by_name(files):
...
+            self.disks = sorted(re.findall(r"[A-Z]+:.*$", os.popen("mountvol /").read(), re.MULTILINE))
+        elif platform == 'linux' or platform == 'android':
```
Comment (HeaTTheatR) [inline kivymd/uix/filemanager.py]: `Linux`, `Mac OS`, `Android` == `Unix`.
So the below code will be correct:

```python
elif platform in ["linux", "android", "macosx"]:
    ...
```

**Comment by HeaTTheatR** [comment]
When you go back to the root directory, the directory list doubles:

Без имени

**Comment by Neizvestnyj** [comment]
> When you go back to the root directory, the directory list doubles:
> 
> Без имени

I will fix it soon

File: kivymd/uix/filemanager.py (hunk 3 above)
Comment (Neizvestnyj) [reply kivymd/uix/filemanager.py]: Yes)

**Comment by Neizvestnyj** [comment]
@HeaTTheatR All your comments are taken into account

**Comment by HeaTTheatR** [comment]
> @HeaTTheatR All your comments are taken into account

But the drive list is still duplicated...

**Comment by Neizvestnyj** [comment]
> > @HeaTTheatR All your comments are taken into account
> 
> But the drive list is still duplicated...

Hmm, I tested everything on **WIndows** and **Linux**, everything works

**Comment by Neizvestnyj** [comment]
> > @HeaTTheatR All your comments are taken into account
> 
> But the drive list is still duplicated...


https://user-images.githubusercontent.com/40869738/121789841-25abde00-cbe2-11eb-826d-47fba3d4baa9.mp4

**Comment by HeaTTheatR** [comment]
@Neizvestnyj I tested again on Mac OS. It works.

**Pull request merged**